packages/two_dimensional_scrollables/lib/src/table_view/table.dart

```diff
@@ -992,9 +1305,13 @@ class RenderTableViewport extends RenderTwoDimensionalViewport {
     // Cells
     for (int column = leading.column; column <= trailing.column; column++) {
       for (int row = leading.row; row <= trailing.row; row++) {
-        final RenderBox cell = getChildFor(
-          TableVicinity(column: column, row: row),
-        )!;
+        final TableVicinity vicinity = TableVicinity(column: column, row: row);
+        final RenderBox? cell = getChildFor(vicinity, allowMerged: false);
+        if (cell == null) {
+          // Covered by a merged cell
+          assert(_mergedVicinities.keys.contains(vicinity));
```

Comment:
hello @Piinks , I was testing out this PR and came across this line failing, when cells are not visible

to reproduce:

```dart
TableView(
    delegate: TableCellBuilderDelegate(
     columnCount: 20, // should be a high enough number for the screen to scroll horizontally
      rowCount: 10, // doesn't matter
      cellBuilder: (context, vicinity) {
        // simulate a table where all cells are merged except in the first row
        final canMerge =
            vicinity.row == 0 ? false : vicinity.column <= colCount - 2;
        if (canMerge && vicinity.column % 2 != 0) {
          //merge every two cells
          return null;
        }
        return TableViewCell(
          columnMergeStart: canMerge ? vicinity.column : null,
          columnMergeSpan: canMerge ? 2 : null,
          child: Container(
            decoration: BoxDecoration(
              color: Colors.red,
              border: Border.all(
                color: Colors.black,
                width: 1,
              ),
            ),
            child: Text('Row: ${vicinity.row}, Column: ${vicinity.column}'),
          ),
        );
      },
      columnBuilder: (index) {
        return const TableSpan(
          extent: FixedTableSpanExtent(100),
        );
      },
      rowBuilder: (index) {
        return const TableSpan(
          extent: FixedTableSpanExtent(100),
        );
      },
    ),
  )
```

Initially when the screen is showed, everything is working fine, but then you scroll enough to the right to hide the first column on the left, and you see this assert fails:


> _mergedVicinities.keys.contains(vicinity)
> is not true

when debugging the assertion, vicinity is at value `(row: 1, column: 1)`, and indeed it doesn't exist in the `_mergedVicinities`, which is understandable since it's hidden.

so for some reason this code is trying to paint a hidden cell

Also thanks for the great work on this package and 2d scrolling in general !

Reply:
Here is what the screen looks like without scrolling:
![image](https://github.com/flutter/packages/assets/63286031/3ea13a66-c2f4-46f5-88ae-1a903711bc02)

if you lower the width, and scroll to the right, the assert fails

Reply:
Thanks for taking a look @Amir-P! Can you share a fully runnable sample of the above code? I do not know what colCount is for example.

Reply:
@Piinks Sure, it's just a simple scaffold

```dart
import 'package:flutter/material.dart';
import 'package:two_dimensional_scrollables/two_dimensional_scrollables.dart';

const colCount = 10;

class TwoDimensionalPage extends StatelessWidget {
  const TwoDimensionalPage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("2D scrolling gantt"),
      ),
      body: TableView(
        delegate: TableCellBuilderDelegate(
          columnCount: colCount,
          rowCount: 10,
          cellBuilder: (context, vicinity) {
            final canMerge =
                vicinity.row == 0 ? false : vicinity.column <= colCount - 2;
            if (canMerge && vicinity.column % 2 != 0) {
              //merge every two cells
              return null;
            }
            return TableViewCell(
              columnMergeStart: canMerge ? vicinity.column : null,
              columnMergeSpan: canMerge ? 2 : null,
              child: Container(
                decoration: BoxDecoration(
                  color: Colors.red,
                  border: Border.all(
                    color: Colors.black,
                    width: 1,
                  ),
                ),
                child: Text('Row: ${vicinity.row}, Column: ${vicinity.column}'),
              ),
            );
          },
          columnBuilder: (index) {
            return const TableSpan(
              extent: FixedTableSpanExtent(100),
            );
          },
          rowBuilder: (index) {
            return const TableSpan(
              extent: FixedTableSpanExtent(100),
            );
          },
        ),
      ),
    );
  }
}
```

Reply:
Ah I see an issue in the code actually. Returning null is not correct for building merged cells.

The same cell, with the same merge config must be returned for every vicinity that the merged cell contains. This is because the table is built lazily, and it is possible for only part of the merged cell to be visible. So here when the merged cell spanning (1,0) and (1,1) is scrolled away, such that (1,0) is not visible, the table calls on (1,1) to build the cell. Right now it looks like the code returns null.

Also, the `columnMergeStart` should be the same for every cell in the merged span. The code shows that it will be `vicinity.column`, which would also cause a crash or unexpected behavior. `columnMergeStart` should always reflect the leading vicinity, not the vicinity that was called by the builder, since any part of the merged cell could be visible.

For example, let's say (1,0) and (1,1) are merged.
These should both be configured with the same merge start and span value.
`columnMergeStart: 1,`
`columnMergeSpan: 2`
This basically tells the viewport, when one of these are built, the merged portion starts at 1 and spans 2 columns.

The viewport will only build one, since once it gets the information, it does not need to build more for the merged span.

So when the app starts, it calls (1,0), and lays out as expected.

When (1,0) is scrolled out of view, but (1,1) is still visible, the viewport calls build for (1,1), which contains the same information that is needed to ay out the merged cell properly.

I really appreciate you taking a look! I am going to add to the error message of that assertion, and some sample code. I think a diagram of some sort would also be beneficial. I might do some of this in a separate change though, since this is already quite large.